src/mon/FSCommands.cc

```diff
@@ -248,6 +248,25 @@ class FsNewHandler : public FileSystemCommandHandler
       }
     }
 
+    vector<string> fsops_vec;
+    cmd_getval(cmdmap, "set", fsops_vec);
+    if(!fsops_vec.empty()) {
+      if(fsops_vec[0] != "set") {
+        ss << "invalid command";
+        return -EINVAL;
+      }
+      if(fsops_vec.size() % 2 == 0 || fsops_vec.size() < 2) {
+      /* since "set" is part of fs options vector, if size of vec is divisble
+      by 2, it indicates that the fsops key-value pairs are incomplete e.g.
+      ["set", "max_mds", "2"]   # valid
```

Comment:
Allow `["set"]`. It makes building command lines easier. i.e. no setting is valid.

Reply:
you mean `ceph fs new ... set` should not throw any error?

Reply:
yes.

Reply:
by no error, you mean to allow it create fs too?

Reply:
yes

Reply:
allowing `foo` fs to be created at `ceph fs new foo foo.meta foo.data set` looks not so good;  i mean its just like what we discussed earlier where I allowed FS creation even if setting fs ops failed. if `set` is to be used, it needs to be provided fs ops key-val pairs i feel.

Reply:
If you're building `fs new` programmatically where 0 or more settings may be applied, it's simpler if `set` can be appended unconditionally. Then, any settings can be appended if present. For example:

```
args = ['fs', 'new']
args += ['some name']
args += ['set']
if allow_standby_replay:
    args += ['allow_standby_replay', 'true']
# etc.
```

make sense?

Reply:
it does but someone in future is going to create a tracker that says "fs new allows fs creation with incomplete just set". allowing `ceph fs new foo foo.meta foo.data set` execute successfully looks more of a bug on CLI; also it's the way it's coded, i.e. if the param is of `type=CephString,n=N`, then even the name of that param is included in the vector, if we want then maybe it is better to change that, i.e. only add elements following param name to vector and not the param name too.